dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -1264,8 +1262,26 @@ synchronized String evaluateInterval() {
         long deltaCompleted = completedWork - previousCompleted;
         double throughput = 1000.0 * deltaCompleted / deltaTime;
         try {
-            int queueDepth = threadPool.getQueue().size();
-            boolean queueEmpty = (queueDepth == 0);
+            queueDepth = threadPool.getQueue().size();
+            boolean queueEmpty = (queueDepth <= 0);
+
+            // Count the number of consecutive times we've seen an empty queue
+            if (!queueEmpty) {
+                consecutiveQueueEmptyCount = 0;
+            } else if (lastAction != LastAction.SHRINK) { // 9/5/2012
```

Comment:
good catch ... the hang resolution call should be to adjustPoolSize not setPoolSize, adjustPoolSize handles setting LastAction appropriately

checkTargetPoolSize is not an issue, it only does setPoolSize if some prior pool size change has failed to complete for some reason, and the prior pool size change should have been driven thru adjustPoolSize which will have set LastAction ... does that make sense?

fix for hang resolution pool size adjustment pushed

Reply:
I am thinking that if checkTargetPoolSize returns true that it should set lastAction to NONE or to GROW depending on what it does.

Reply:
handleOutliers also does an action possibly.  It may do NONE or it may do SHRINK I am thinking.

Reply:
We don't change pool size at all in handleOutliers, though it may do a reset of the distribution data for the pool size or a reset of the whole threadpool ... in either case, I think that it is probably OK (maybe even preferable?) to leave LastAction as it was set in the prior (non-outlier) cycle, rather than changing it in response to weird/outlier data